File: back-end/Back-End/Models/CrPlayer.cs
using Back_End.Models.CrModels;

namespace Back_End.Models
{
    public class CrPlayer
    {
        public string tag { get; set; }
        public string name { get; set; }
        public int expLevel { get; set; }
        public int trophies { get; set; }
        public int bestTrophies { get; set; }
        public int wins { get; set; }
        public int losses { get; set; }
        public int battleCount { get; set; }
        public int threeCrownWins { get; set; }
        public int challengeCardsWon { get; set; }
        public int challengeMaxWins { get; set; }
        public int tournamentCardsWon { get; set; }
        public int tournamentBattleCount { get; set; }
        public int donations { get; set; }
        public int donationsReceived { get; set; }
        public int totalDonations { get; set; }
        public int warDayWins { get; set; }
        public int clanCardsCollected { get; set; }
        public string role { get; set; }
        public CrPlayerClan clan { get; set; }
        public Arena arena { get; set; }
        public PlayerLeagueStatistics leagueStatistics { get; set; }
        public List<CrPlayerItemLevel> cards { get; set; }
        public List<CrPlayerItemLevel> currentDeck { get; set; }
        public Item currentFavouriteCard { get; set; }
        public int starPoints { get; set; }
        public int expPoints { get; set; }
        public int totalExpPoints { get; set; }
        public PathOfLegendSeasonResult currentPathOfLegendSeasonResult { get; set; }
        public PathOfLegendSeasonResult lastPathOfLegendSeasonResult { get; set; }
        public PathOfLegendSeasonResult bestPathOfLegendSeasonResult { get; set; }
    }
}

File: front-end/my-app/src/models/CrPlayer.ts
export interface CocPlayer {
    tag: string,
    name: string,
    expLevel: number,
    trophies: number,
    bestTrophies: number,
    wins: number,
    losses: number,
    battleCount: number,
    threeCrownWins: number,
    challengeCardsWon: number,
    challengeMaxWins: number,
    tournamentCardsWon: number,
    tournamentBattleCount: number,
    donations: number,
    donationsReceived: number,
    totalDonations: number,
    warDayWins: number,
    clanCardsCollected: number,
    role: string,
    clan: CrPlayerClan | undefined,
    arena: Arena | undefined,
    leagueStatistics: PlayerLeagueStatistics | undefined,
    cards: Array<CrPlayerItemLevel>,
    currentDeck: Array<CrPlayerItemLevel>,
    currentFavouriteCard: Item | undefined,
    starPoints: number,
    expPoints: number,
    totalExpPoints: number,
    currentPathOfLegendSeasonResult: PathOfLegendSeasonResult | undefined,
    lastPathOfLegendSeasonResult: PathOfLegendSeasonResult | undefined,
    bestPathOfLegendSeasonResult: PathOfLegendSeasonResult | undefined,
}

interface CrPlayerClan {
    name: string,
    tag: string,
    badgeId: number,
    badgeUrls: {
        [key: string]: Dict
    }
}

interface Arena {
    id: number,
    name: string,
    iconUrls: {
        [key: string]: Dict
    }
}

interface PlayerLeagueStatistics {
    currentSeason: LeagueSeasonResult,
    previousSeason: LeagueSeasonResult,
    bestSeason: LeagueSeasonResult,
}

interface LeagueSeasonResult {
    id: string,
    rank: number,
    trophies: number,
    bestTrophies: number,
}

interface CrPlayerItemLevel {
    name: string,
    id: number,
    level: number,
    starLevel: number,
    maxLevel: number,
    evolutionLevel: number,
    maxEvolutionLevel: number,
    count: number,
    used: boolean,
    iconUrls: {
        [key: string]: Dict
    }
}

interface Item {
    name: string,
    id: number,
    evolutionLevel: number,
    maxEvolutionLevel: number,
    iconUrls: {
        [key: string]: Dict
    }
}

interface PathOfLegendSeasonResult {
    leagueNumber: number,
    trophies: number,
    rank: number,
}

interface Dict {
    additionalProp1: string,
    additionalProp2: string,
    additionalProp3: string,
}
